src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js

```diff
@@ -1,161 +1,143 @@
-(function() {
-  'use strict';
+(function () {
+    'use strict';
 
-  function ListViewSettingsDirective(dataTypeResource, dataTypeHelper, listViewPrevalueHelper) {
+    function ListViewSettingsDirective(dataTypeResource, dataTypeHelper, editorService, listViewPrevalueHelper) {
 
-    function link(scope) {
+        function link(scope) {
 
-      scope.dataType = {};
-      scope.editDataTypeSettings = false;
-      scope.customListViewCreated = false;
+            scope.dataType = {};
+            scope.customListViewCreated = false;
+            
+            const checkForCustomListView = () => scope.dataType.name === "List View - " + scope.modelAlias;
 
-      /* ---------- INIT ---------- */
+            /* ---------- INIT ---------- */ 
 
-      function activate() {
+            const activate = () => {
 
-        if(scope.enableListView) {
+                if (scope.enableListView) {
 
-          dataTypeResource.getByName(scope.listViewName)
-            .then(function(dataType) {
+                    dataTypeResource.getByName(scope.listViewName)
+                        .then(dataType => {
 
-              scope.dataType = dataType;
+                            scope.dataType = dataType;
 
-              listViewPrevalueHelper.setPrevalues(dataType.preValues);
-              scope.customListViewCreated = checkForCustomListView();
-            });
-
-        } else {
-
-          scope.dataType = {};
-
-        }
-
-      }
-
-      /* ----------- LIST VIEW SETTINGS --------- */
-
-      scope.toggleEditListViewDataTypeSettings = function() {
-        scope.editDataTypeSettings = !scope.editDataTypeSettings;
-      };
-
-      scope.saveListViewDataType = function() {
-
-          var preValues = dataTypeHelper.createPreValueProps(scope.dataType.preValues);
-
-          dataTypeResource.save(scope.dataType, preValues, false).then(function(dataType) {
-
-              // store data type
-              scope.dataType = dataType;
+                            listViewPrevalueHelper.setPrevalues(dataType.preValues);
+                            scope.customListViewCreated = checkForCustomListView();
+                        });
 
-              // hide settings panel
-              scope.editDataTypeSettings = false;
+                } else {
+                    scope.dataType = {};
+                } 
+            } 
 
-          });
+            /* ----------- LIST VIEW SETTINGS --------- */
+            const showSettingsOverlay = () => {
+                const overlay = {
+                    view: 'views/components/umb-list-view-settings-overlay.html',
+                    hideDescription: true,
+                    hideIcon: true,
+                    size: 'small',
```

Comment:
I think it should be in the large infinite editor, that would give the settings some space to breathe.

Reply:
Hi @skttl, missed your comment - is there an inbetween size? The default (per infinite editor for a document type) might be too wide, something inbetween would be goldilocks-just-right. Can add an umb-editor--medium option easily enough

Reply:
No, only large and small is supported. But maybe a PR with a medium setting would be approved? :)

Reply:
Looks like I never replied to this one : https://github.com/umbraco/Umbraco-CMS/issues/5516 :-o

But yes, we'd be good with adding a medium!

Reply:
One medium coming up. Will keep it separate from this PR though.

If I update my local branch from upstream/v8/dev, will I end up with a heap of additional git history/commits? This branch was 8.1, so it's stale.

Reply:
You will, but they'll be in your local v8/dev branch. Shouldn't be a problem! And you could even merge v8/dev into this PR branch as well.

Reply:
I've accepted the `medium` PR and updated this PR. Also merged with v8/dev which went well.